### PR TITLE
tests: teach tests boilerplate to run selected tests only

### PR DESCRIPTION
**Pull request by poettering**
sometimes its useful to only run a specific test (or multiple) instead of all implemented in a test. Allow the test name to be specified on the command line.

### CODE REVIEW

**Comment by mrc0mmand** [comment]
This seems to break TEST-35-LOGIN:

```
00:24:01.095 [   74.338754] testsuite-35.sh[1499]: ++ loginctl list-sessions --no-legend
00:24:01.095 [   74.339601] testsuite-35.sh[1500]: ++ awk '$3 == "logind-test-user" { print $1 }'
00:24:01.095 [   74.346054] testsuite-35.sh[1416]: + s=10
00:24:01.095 [   74.346125] testsuite-35.sh[1416]: + /usr/lib/systemd/tests/unit-tests/manual/test-session-properties /org/freedesktop/login1/session/_310 /dev/tty2
00:24:01.095 [   74.358569] testsuite-35.sh[1501]: No matching tests found.
```

File: src/shared/tests.h [hunk 1]
```diff
@@ -82,15 +83,19 @@ extern const TestFunc _weak_ __stop_SYSTEMD_TEST_TABLE[];
...
+        bool ran = false;
 
         if (!__start_SYSTEMD_TEST_TABLE)
                 return r;
 
         const TestFunc *t = ALIGN_PTR(__start_SYSTEMD_TEST_TABLE);
         while (t < __stop_SYSTEMD_TEST_TABLE) {
 
+                if (!strv_isempty(tests) && !strv_contains(tests, t->name))
+                        goto skip;
```
Comment (dtardon) [inline src/shared/tests.h]: It looks this is in fact a `for` loop, isn't it? And if it were written as a `for` loop, you could just use `continue` here.

**Comment by poettering** [comment]
So apparently the command line is already used by some tests for other purposes. Hence let's just make it an env var instead. PTAL.

**Review by mrc0mmand — APPROVED** [review]

**Comment by bluca** [comment]
can the env var be documented somewhere? maybe in test/README?

**Comment by poettering** [comment]
> can the env var be documented somewhere? maybe in test/README?

→ #28124

**Pull request merged**